components/brave_wallet/browser/brave_wallet_utils_unittest.cc

```diff
@@ -1187,4 +1187,67 @@ TEST(BraveWalletUtilsUnitTest, GetSnsRpcUrl) {
   EXPECT_EQ(GURL("https://mainnet-beta-solana.brave.com/rpc"), GetSnsRpcUrl());
 }
 
+TEST(BraveWalletUtilsUnitTest, GetChainIdByNetworkId) {
+  TestingPrefServiceSimple prefs;
+  prefs.registry()->RegisterDictionaryPref(kBraveWalletCustomNetworks);
+  prefs.registry()->RegisterBooleanPref(kSupportEip1559OnLocalhostChain, false);
+
+  for (auto coin :
+       {mojom::CoinType::ETH, mojom::CoinType::FIL, mojom::CoinType::SOL}) {
+    ASSERT_TRUE(GetAllCustomChains(&prefs, coin).empty());
+    std::vector<base::Value::Dict> values;
+    mojom::NetworkInfo chain1 = GetTestNetworkInfo1();
+    chain1.coin = coin;
+    values.push_back(NetworkInfoToValue(chain1));
+    mojom::NetworkInfo chain2 = GetTestNetworkInfo2();
+    chain2.coin = coin;
+    if (coin != mojom::CoinType::ETH) {
+      chain2.is_eip1559 = false;
+    }
+    values.push_back(NetworkInfoToValue(chain2));
+    UpdateCustomNetworks(&prefs, std::move(values), coin);
+  }
+
+  auto getChainIdByNetworkIdCheck = [&](const mojom::CoinType& coin_type) {
+    auto chains = GetAllChains(&prefs, coin_type);
+    const auto& custom_networks = prefs.GetDict(kBraveWalletCustomNetworks);
+    auto* custom_list =
+        custom_networks.FindList(GetPrefKeyForCoinType(coin_type));
+    if (custom_list) {
+      for (const auto& it : *custom_list) {
+        chains.push_back(ValueToNetworkInfo(it));
+      }
+    }
```

Comment:
Aren't these custom networks in GetAllChains result already? What's the reason for adding them again?

Reply:
I still don't see why we have to manually add results from custom networks pref, `GetAllChains` should already include both known networks and custom networks. (https://github.com/brave/brave-core/blob/6f0d20612c20223a46082a2ea084d5486022fac3/components/brave_wallet/browser/brave_wallet_utils.cc#L1052-L1053)

Reply:
you are right, removed it